hibernate-core/src/main/java/org/hibernate/sql/results/graph/entity/internal/EntityDelayedFetchInitializer.java

```diff
@@ -197,28 +198,33 @@ public void resolveInstance(EntityDelayedFetchInitializerData data) {
 					// because we can't generate a proxy based on the unique key yet
 					if ( referencedModelPart.isLazy() ) {
 						instance = UNFETCHED_PROPERTY;
-					}
-					else if ( getParent().isEntityInitializer() && isLazyByGraph( rowProcessingState ) ) {
-						// todo : manage the case when parent is an EmbeddableInitializer
-						final Object resolvedInstance = getParent().asEntityInitializer()
-								.getResolvedInstance( rowProcessingState );
-						final LazyAttributeLoadingInterceptor persistentAttributeInterceptor = (LazyAttributeLoadingInterceptor) ManagedTypeHelper
-								.asPersistentAttributeInterceptable( resolvedInstance ).$$_hibernate_getInterceptor();
-
-						persistentAttributeInterceptor.addLazyFieldByGraph( navigablePath.getLocalName() );
-						instance = UNFETCHED_PROPERTY;
-					}
-					else {
-						instance = concreteDescriptor.loadByUniqueKey(
-								uniqueKeyPropertyName,
-								data.entityIdentifier,
-								session
-						);
-
-						// If the entity was not in the Persistence Context, but was found now,
-						// add it to the Persistence Context
-						if ( instance != null ) {
-							persistenceContext.addEntity( euk, instance );
+					} else {
+						// Try to load a PersistentAttributeInterceptable. If we get one, we can add the lazy
+						// field to the interceptor. If we don't get one, we load the entity by unique key.
+						PersistentAttributeInterceptable persistentAttributeInterceptable = null;
+						if (getParent().isEntityInitializer() && isLazyByGraph( rowProcessingState )) {
```

Comment:
Please follow Hibernate's [code style](https://hibernate.org/community/contribute/intellij-idea/) guidelines.

```suggestion
						if ( getParent().isEntityInitializer() && isLazyByGraph( rowProcessingState ) ) {
```

Reply:
Well, using "commit suggestion" threw the PR into a weird state because of commit messages. Sorry for all the force pushes.